src/main/java/slimeknights/tconstruct/tools/modifiers/ability/GlowingModifier.java

```diff
@@ -0,0 +1,31 @@
+package slimeknights.tconstruct.tools.modifiers.ability;
+
+import net.minecraft.item.ItemUseContext;
+import net.minecraft.util.ActionResultType;
+import slimeknights.tconstruct.library.modifiers.SingleUseModifier;
+import slimeknights.tconstruct.library.tools.nbt.IModifierToolStack;
+import slimeknights.tconstruct.shared.TinkerCommons;
+
+public class GlowingModifier extends SingleUseModifier {
+  public GlowingModifier() {
+    super(0xffffaa);
+  }
+
+  @Override
+  public int getPriority() {
+    return 180;
+  }
+  
+  @Override
+  public ActionResultType onItemUse(IModifierToolStack tool, int level, ItemUseContext context) {
+    if (tool.getCurrentDurability() >= 5 && context.getPlayer().isSneaking()) {
+      if (!context.getWorld().isRemote) {
+        if (TinkerCommons.glow.get().addGlow(context.getWorld(), context.getPos(), context.getFace().getOpposite())) {
+          tool.setDamage(tool.getDamage() + 5);
+          }
+        }
+      return ActionResultType.func_233537_a_(context.getWorld().isRemote);
+    }

```

Comment:
Looks like this is targeting the wrong position, should offset backwards by the face. Also, `getPlayer()` is nulllable on the context. The snippet below addresses those issues and adds a sound effect.

```java
    PlayerEntity player = context.getPlayer();
    if (tool.getCurrentDurability() >= 5 && (player == null || player.isSneaking())) {
      if (!context.getWorld().isRemote) {
        World world = context.getWorld();
        Direction face = context.getFace();
        BlockPos pos = context.getPos().offset(face);
        if (TinkerCommons.glow.get().addGlow(world, pos, face.getOpposite())) {
          tool.setDamage(tool.getDamage() + 5);
          world.playSound(null, pos, world.getBlockState(pos).getSoundType(world, pos, player).getPlaceSound(), SoundCategory.BLOCKS, 1.0f, 1.0f);
        }
      }
      return ActionResultType.func_233537_a_(context.getWorld().isRemote);
    }
 ```